### PR TITLE
fragments: 2.0.2 -> 2.1; unbreak

### LINKED ISSUE #217170: Update request: fragments 2.0.2 → 2.1
- Package name: fragments
- Latest released version: [2.1](https://gitlab.gnome.org/World/Fragments/-/releases/2.1)

- Current version on the unstable channel: 2.0.2
- Current version on the stable/release channel: 2.0.2

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@emilytrau 

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by chuangzhu**
###### Description of changes

Fixes #217170 #218310

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/p2p/fragments/default.nix [hunk 1]
```diff
@@ -31,24 +30,23 @@ let
...
 
-  postPatch = ''
-    patchShebangs build-aux/meson/postinstall.py
-  '';
-
+  # https://github.com/gtk-rs/gtk4-rs/issues/1201
+  patches = [ ./gtk4-rs.patch ];
   cargoDeps = rustPlatform.fetchCargoTarball {
     inherit src;
+    patches = [ ./gtk4-rs.patch ];
```
Comment (NickCao) [inline pkgs/applications/networking/p2p/fragments/default.nix:47]: Or just inherit patches?

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 220618` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fragments</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]
Tested downloading sample torrent.

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 220618` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fragments</li>
  </ul>
</details>

**Pull request merged**